archiver/archive.py

```diff
@@ -158,13 +165,17 @@ def update_archive(archive_path, links, source=None, resume=None, append=True):
         source = download_url(source)
 
     # Step 1: Parse the links and dedupe them with existing archive
-    links = merge_links(archive_path=out_dir, import_path=source)
-    
+    links = merge_links(archive_path=out_dir, import_path=source, only_new=False)
+    new_links = merge_links(archive_path=out_dir, import_path=source, only_new=True)
+
     # Step 2: Write new index
     write_links_index(out_dir=out_dir, links=links)
```

Comment:
I think you're missing `links=new_links` here, otherwise the index wont be updated to include the new links.

Reply:
No, that is fine. `links` contain all links, including the new one. `new_links` only include the new links, so the newly generated index would contain only the new links.